### PR TITLE
OCR2Keeper: Implement pipeline transmitter 

### PR DESCRIPTION
**Pull request by begmaroman**
As a node operator, I would like to see records when the ocr2keeper job transmits the performance. 

The default pipeline runner could be used on the transmission step so the job run would be automatically inserted into the database.

### CODE REVIEW

File: core/services/relay/evm/evm.go [hunk 1]
```diff
@@ -134,30 +135,54 @@ func newConfigProvider(lggr logger.Logger, chainSet evm.ChainSet, args relaytype
 	}, nil
 }
 
-func newContractTransmitter(lggr logger.Logger, rargs relaytypes.RelayArgs, transmitterID string, configWatcher *configWatcher) (*ContractTransmitter, error) {
+func newContractTransmitter(lggr logger.Logger, rargs relaytypes.RelayArgs, transmitterID string, configWatcher *configWatcher, pr pipeline.Runner) (*ContractTransmitter, error) {
```
Comment (infiloop2) [inline core/services/relay/evm/evm.go:138]: not sure if this behaviour is obvious, that depending on pr being nil or not different transmitters are created. wdyt about a separate function all together `newPipelineContractTransmitter` ?

**Review by infiloop2 — COMMENTED** [review]
Changes look good to me. Could you add some tests for it (maybe similar to transmitter_default_test.go)

File: core/services/relay/evm/evm.go (hunk 1 above)
Comment (begmaroman) [reply core/services/relay/evm/evm.go]: Yeah, looks not really good, fixed

**Review by infiloop2 — APPROVED** [review]

File: core/services/ocrcommon/transmitter_pipeline.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+)
+
+const txObservationSource = `
+    transmit_tx [type=ethtx
```
Comment (connorwstein) [inline core/services/ocrcommon/transmitter_pipeline.go:15]: ah is this just to show the eth tx in the UI? I thought there was already an eth transactions page

**Review by EasterTheBunny — APPROVED** [review]

File: core/services/ocrcommon/transmitter_pipeline.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		ID:                t.spec.ID,
+		DotDagSource:      txObservationSource,
+		CreatedAt:         t.spec.CreatedAt,
+		MaxTaskDuration:   t.spec.MaxTaskDuration,
+		GasLimit:          &t.spec.GasLimit.Uint32,
+		ForwardingAllowed: t.spec.ForwardingAllowed,
+		JobID:             t.spec.PipelineSpecID,
+		JobName:           t.spec.Name.ValueOrZero(),
+		JobType:           t.spec.Type.String(),
+	}, vars)
```
Comment (infiloop2) [inline core/services/ocrcommon/transmitter_pipeline.go:85]: setting the obs source on spec might be a simpler way
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/keeper/upkeep_executer.go#L246-L247

**Review by infiloop2 — DISMISSED** [review]
looks good. one minor suggestion

File: core/services/ocrcommon/transmitter_pipeline.go (hunk 3 above)
Comment (begmaroman) [reply core/services/ocrcommon/transmitter_pipeline.go:85]: Basically, this is the common transmitter that could be used for any kind of job. This logic should do exactly the same as the normal transmitter does so I would not delegate the observation source definition to the specific job level but would define it exactly within the pipeline transmitter.

**Review by infiloop2 — APPROVED** [review]

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**